### PR TITLE
FEATURE: Docker connection option added

### PR DESCRIPTION
**Pull request by vikramdevtron**
Related PR:
https://github.com/devtron-labs/devtron/pull/750

Issues fixed:
https://github.com/devtron-labs/devtron/issues/720
https://github.com/devtron-labs/devtron/issues/721

### CODE REVIEW

File: DockerUtil.go [hunk 1]
```diff
@@ -37,13 +33,49 @@ import (
...
-	log.Println(string(out))
-	waitForDockerDaemon(retryCount)
+func StartDockerDaemon(dockerConnection, dockerRegistryUrl, dockerCert string) {
+	connection := dockerConnection
+	u, err := url.Parse(dockerRegistryUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if connection == insecure {
+		dockerdstart := "dockerd --insecure-registry " + u.Host + " --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &"
```
Comment (nishant-d) [inline DockerUtil.go:50]: use string fmt

File: DockerUtil.go [hunk 2]
```diff
@@ -37,13 +33,49 @@ import (
...
+		out, _ := exec.Command("/bin/sh", "-c", dockerdstart).Output()
+		logStage("Insecure Registry")
+		log.Println(string(out))
+		waitForDockerDaemon(retryCount)
+	} else {
+		if connection == secureWithCert {
+			os.MkdirAll("/etc/docker/certs.d/"+u.Host, os.ModePerm)
```
Comment (nishant-d) [inline DockerUtil.go:57]: use string fmt

**Review by nishant-d — CHANGES_REQUESTED** [review]

File: DockerUtil.go [hunk 3]
```diff
@@ -37,13 +33,49 @@ import (
...
+	u, err := url.Parse(dockerRegistryUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if connection == insecure {
+		dockerdstart := fmt.Sprintf("dockerd --insecure-registry %s --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &", u.Host)
+		out, _ := exec.Command("/bin/sh", "-c", dockerdstart).Output()
+		logStage("Insecure Registry")
+		log.Println(string(out))
+		waitForDockerDaemon(retryCount)
```
Comment (manish-agrawal-ai) [inline DockerUtil.go:54]: This logic can be extracted out in a method as same is being used in else condition

**Review by manish-agrawal-ai — COMMENTED** [review]
check comment

**Review by nishant-d — APPROVED** [review]

**Pull request merged**